packages/cursorless-vscode/src/ide/vscode/VscodeRunMode.ts

```diff
@@ -8,5 +8,29 @@ const EXTENSION_MODE_MAP: Record<ExtensionMode, RunMode> = {
 };
 
 export function vscodeRunMode(extensionContext: ExtensionContext): RunMode {
-  return EXTENSION_MODE_MAP[extensionContext.extensionMode];
+  const envMode = getEnvironmentalMode();
+  const extensionMode = EXTENSION_MODE_MAP[extensionContext.extensionMode];
+
+  if (envMode !== extensionMode) {
+    throw new Error(
+      `Extension mode '${extensionMode}' doesn't match environmental mode '${envMode}'`,
+    );
+  }
+
+  return extensionMode;
+}
+
+function getEnvironmentalMode(): RunMode {
```

Comment:
this should probably be moved outside of cursorless-vscode so it can be used by neovim

Reply:
Yes this should probably move to a node package

Reply:
Or should be us rename our existing file system package to node? I know @pokey was leaning that way originally.

Reply:
done